src/arraymath.jl

```diff
@@ -149,3 +149,4 @@ end
         return a
     end
 end
+arithmetic_closure(T) = typeof((one(T)*zero(T) + zero(T))/one(T))
```

Comment:
Could be mysterious for someone in the future.  Might be worth a one line explanation or so.